### PR TITLE
chore: Add github action to run e2e tests in k3s

### PR DESCRIPTION
**Pull request by jessesuen**
GitHub action to:
* start k3s
* start the controller
* run e2e tests

### CODE REVIEW

File: rollout/controller.go [hunk 1]
```diff
@@ -341,15 +341,14 @@ func (c *Controller) syncHandler(key string) error {
...
-	log.WithField(logutil.RolloutKey, name).WithField(logutil.NamespaceKey, namespace).Infof("Started syncing rollout at (%v)", startTime)
 	rollout, err := c.rolloutsLister.Rollouts(namespace).Get(name)
 	if k8serrors.IsNotFound(err) {
-		log.WithField(logutil.RolloutKey, name).WithField(logutil.NamespaceKey, namespace).Info("Rollout has been deleted")
 		return nil
 	}
 	if err != nil {
 		return err
 	}
+	log.WithField(logutil.RolloutKey, name).WithField(logutil.NamespaceKey, namespace).Infof("Started syncing rollout at (%v)", startTime)
```
Comment (jessesuen) [inline rollout/controller.go:351]: This was previously logging rollouts that were enqueued that were not matching the instance id.

File: test/e2e/canary_test.go [hunk 2]
```diff
@@ -23,7 +23,7 @@ func (s *CanarySuite) TestCanarySetCanaryScale() {
 - setWeight: 50
 - pause: {}
 - setCanaryScale:
-    replicas: 6
+    replicas: 3
```
Comment (jessesuen) [inline test/e2e/canary_test.go:26]: makes test run faster.

File: controller/controller.go [hunk 3]
```diff
@@ -213,6 +213,7 @@ func NewManager(
 	})
 
 	ingressController := ingress.NewController(ingress.ControllerConfig{
+		Client:           kubeclientset,
```
Comment (jessesuen) [inline controller/controller.go:216]: This was a bug causing ALB implementation to panic

**Review by khhirani — APPROVED** [review]
LGTM

**Pull request merged**